### PR TITLE
fix: Update diffDOM to the latest version

### PR DESCRIPTION
**Pull request by petrklus**
## Description

Work in progress on https://github.com/django-cms/django-cms/issues/7460. 

Upgrading diffDOM seems to resolve the issue, however, is not compatible with UglifyJS. 

## Related resources

The issue seems to be related to variable naming in diffDOM:

image

I belive it is due to the variable called `node`:

https://github.com/fiduswriter/diffDOM/blob/v4.2.8/src/diffDOM/dom/fromVirtual.js#L2

Which UglifyJS does not like. I am seeking help with getting this fix across the finish line.

## Checklist

* [x] I have opened this pull request against ``develop``
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.
* [ ] I have added or modified the tests when changing logic
* [ ] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
   },
   "dependencies": {
     "babel-runtime": "^6.23.0",
-    "diff-dom": "2.5.1",
+    "diff-dom": "4.2.8",
```
Comment (DmytroLitvinov) [inline package.json:52]: Should we add it with `^` sign to allow upgrade for minor version in future in package-lock.json file?

**Comment by fsbraun** [comment]
@vinitkumar Do you have an idea why the javascript tests fail?

**Comment by petrklus** [comment]
@fsbraun @vinitkumar have you been able to get the bottom of the tests failing?

**Comment by fsbraun** [comment]
So far, I have not succeeded. @FinalAngel Do you think you can take a look?

**Comment by fsbraun** [comment]
@petrklus It seems I have closed this PR. I am not certain why and how. Also, I cannot reopen. Can you reopen this PR?

**Comment by fsbraun** [comment]
OMG, I did not do this on purpose! It's restored now. Is there a particular reason you opened the PR against `release/3.11.x`?  The better way would be to open it against `develop`.

**Comment by vinitkumar** [comment]
We will close this PR once https://github.com/django-cms/django-cms/pull/7600 is merged as it does the same thing, but also got the build running. 

Also, @petrklus, I added you as a co-author of the commit to properly attribute you for the fix and the initiative. Thanks for helping us figure this one out and of-course of reporting this bug.

**Comment by fsbraun** [comment]
Fixed by #7600. Thank you, @petrklus, for providing this solution!

**Pull request closed without merging**